src/background.js

```diff
@@ -19,3 +23,117 @@ chrome.runtime.onMessage.addListener((request) => {
     chrome.runtime.openOptionsPage();
   }
 });
+
+// Message listener from content scripts
+chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+  // do not attempt to parse the message if it is not coming from a content script
+  if (!sender.tab) {
+    return;
+  }
+
+  switch (request.operation) {
+    case "LOAD_WORKSPACES":
+      loadWorkspaces(request).then(sendResponse);
+      break;
+    case "GET_OAUTH_TOKEN":
+      getOAuthToken(request).then(sendResponse);
+      break;
+    case "REFRESH_OAUTH_TOKEN":
+      refreshOAuthToken(request).then(sendResponse);
+      break;
+
+    default:
+      console.error(`Unknown operation: ${request.operation}`);
+  }
+  return true;
+});
+
+/**
+ * Fetch workspaces from Orbit API
+ * Credentials passed as arguments since this is called from options page
+ *
+ * @param {String} accessToken
+ * @param {String} apiKey
+ *
+ * @returns {success, response, ok}
+ */
+const loadWorkspaces = async ({ accessToken, apiKey }) => {
+  try {
+    const url = new URL(`${ORBIT_API_ROOT_URL}/workspaces`);
+
+    const { params, headers } = configureRequest({
+      ACCESS_TOKEN: accessToken,
+      API_TOKEN: apiKey,
+    });
+
+    url.search = params.toString();
+
+    const response = await fetch(url, {
+      headers: headers,
+    });
+
+    return { success: true, response: await response.json(), ok: response.ok };
```

Comment:
Could we only keep one of {`success`, `ok`}? It looks like they’re essentially the same thing.

Reply:
Not exactly - so `success: true` is replacing the `try/catch` block. IE, if `success: false` then the request has failed, and we can handle the exception. 

`ok: response.ok` is checking if the response succeeded and has a 2XX status. We use this condition to check for requests that succeeded, but were invalid - ie, a 401 unauthorised when logging in. 

Cleaner solution could be returning `status:` instead of `success:` - but might trip us up later with handling authentication errors. AFAIK we would only be worried about 401, but not sure if there's any other codes we should expect.

WDYT?